port/cpl_levenshtein.cpp

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2019, Guilherme Agostinelli
+// Ported from https://github.com/guilhermeagostinelli/levenshtein/blob/master/levenshtein.cpp
+
+#include "cpl_levenshtein.h"
+
+#include <algorithm>
+#include <cstring>
+#include <limits>
+#include <vector>
+
+/** Computes the Levenshtein distance between 2 words.
+ *
+ * If transpositionAllowed = true, then it is the Damerau-Levenshtein distance
+ *
+ * Return SIZE_T_MAX in case of error.
+ */
+size_t CPLLevenshteinDistance(const char *word1, const char *word2,
+                              bool transpositionAllowed)
+{
+    const size_t size1 = strlen(word1);
+    const size_t size2 = strlen(word2);
+
+    // If one of the words has zero length, the distance is equal to the size of the other word.
+    if (size1 == 0)
+        return size2;
+    if (size2 == 0)
+        return size1;
+
+    // Would cause enormous amount of memory to be allocaded
+    if (size1 >= 32768 || size2 >= 32768)
+    {
+        return strcmp(word1, word2) == 0 ? 0
```

Comment:
`size1 + size2` would be more accurate. 

But 2GB is still arguably quite a bit of memory.

Reply:
returning SIZE_T_MAX is on purpose (and documented in the function doc) to indicate that the exact value is unknown